sdk/services/prices.ts

```diff
@@ -219,6 +201,58 @@ export default class PricesService {
 		return offChainPrices;
 	}
 
+	private connectToPyth(networkId: NetworkId, server: PriceServer) {
+		this.pyth = new EvmPriceServiceConnection(getPythNetworkUrl(networkId, server), {
+			logger: LOG_WS ? console : undefined,
+		});
+
+		this.pyth.onWsError = (error) => this.retryConnection(networkId, error);
+
+		this.sdk.context.events.emit('prices_connection_update', {
+			connected: true,
+		});
+
+		return this.pyth;
+	}
+
+	private setEventListeners() {
+		this.sdk.context.events.on('network_changed', (params) => {
+			if (this.pyth) {
```

Comment:
I think we can move this part to the start of `connectToPyth`

Reply:
Done.